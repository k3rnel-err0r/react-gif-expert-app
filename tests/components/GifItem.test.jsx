import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Tests on <GifItem />', () => {
    
    const title = 'Saitama is the best hero';
    const url = 'https://one-punch.com/saitama.jpg';
    
    test('should match snapshot', () => {
        const { container } = render(<GifItem title={ title } url={ url } />);
        expect( container ).toMatchSnapshot();
    });

    test('Should show image with url and alt', () => {
        render(<GifItem title={ title } url={ url } />);
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url);
        // expect( screen.getAllByRole('image').alt).toBe( title );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });

    test('Should show the title in component', () => {
        render(<GifItem title={ title } url={ url } />);
        expect( screen.getByText(title) ).toBeTruthy();
    });

});