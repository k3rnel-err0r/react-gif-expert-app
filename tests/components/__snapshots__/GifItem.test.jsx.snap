// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem /> should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama is the best hero"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama is the best hero
    </p>
  </div>
</div>
`;
